### DISCUSSION TITLE
What is hparams exactly?

### DISCUSSION THREAD

**Discussion by keunwoochoi** [Lightning-AI/pytorch-lightning#5821] [Lightning Trainer API: Trainer, LightningModule, LightningDataModule] [answered]
Hi, thanks for the nice product again.

From #525 and #599, I could guess that `hparams` is required to load a saved model (which I think should be mentioned somewhere in the doc btw). And from the examples, seems like `hparams` may be `argparse.Namespace`. Unfortunately though, it was not so easy to understand the concept. 

What is `hparams` exactly? What kind of information it should/can/should not include to work properly? Is it recommended to use [`hyperparameter argument parser`](https://williamfalcon.github.io/test-tube/hyperparameter_optimization/HyperOptArgumentParser/)? Say, if I'm not into hyperparameter search at the moment and just want to be able to load the checkpoint model, what is the requirement on the hparams?

**Comment by s-rog** [comment]
hparams is just a dict that contains your hyperparameters for easy logging and experiments. It is optional, but does make loading models more tedious.

**Comment by neggert** [comment, marked as answer]
It should be an [`argparse.Namespace`](https://docs.python.org/3.7/library/argparse.html#argparse.Namespace). You can get this from `argparse`, `testtube`'s `HyperOptArgumentParser`, or create it manually from a dict like so: `argparse.Namespace(**my_dict)`.

**Comment by keunwoochoi** [comment]
@s-rog Thanks, but it's not a dict :-) 
@neggert Thanks. Yes those are the info I could *assume* by reading code etc. But I think those and probably more details should be documented somewhere. I'd like to contribute, but I can't document something I don't know clearly. 
Also, in my opinion, it should be very strongly recommended to use it so that they can load the model later, assuming the use cases of `pytorch-lightning` would be more serious than just training something on MNIST. Because, for those toy example, why bother :)

**Comment by s-rog** [comment]
oh yeah I keep forgetting, you're right!

**Comment by neggert** [comment]
Yes, there's a complete overhaul of the docs in progress. We'll make sure this gets in there.

**Comment by williamFalcon** [comment]
thanks for the input! 
Once we have the new docs, please let us know of any problems! :)
